src/lib.rs

```diff
@@ -185,52 +185,78 @@ fn enforce_container_security_policies(
+                                    None,
+                                    None,
+                                )
+                            }
+                        }
+                    }
+                }
+                for container in pod_spec.containers.iter_mut() {
+                    match enforce_container_security_policies(container, &validation_request) {
+                        Ok(mutate_request) => mutated = mutated || mutate_request,
+                        Err(error) => {
+                            return kubewarden::reject_request(
+                                Some(error.to_string()),
+                                None,
+                                None,
+                                None,
+                            )
+                        }
+                    }
+                }
 
-    if pod.spec.is_none() {
-        return kubewarden::accept_request();
-    }
-    let mut pod_spec = pod.spec.unwrap();
-
-    if let Some(init_containers) = pod_spec.init_containers.as_mut() {
-        for init_container in init_containers.iter_mut() {
-            match enforce_container_security_policies(init_container, &validation_request) {
-                Ok(mutate_request) => mutated = mutated || mutate_request,
-                Err(error) => {
-                    return kubewarden::reject_request(Some(error.to_string()), None, None, None)
+                match enforce_supplemental_groups(
+                    pod_spec.security_context.clone(),
+                    &validation_request,
+                ) {
+                    Ok(mutate_security_context) => {
+                        if mutate_security_context.is_some() {
+                            mutated = true;
+                            pod_spec.security_context = mutate_security_context;
+                        }
+                    }
+                    Err(msg) => {
+                        return kubewarden::reject_request(Some(msg.to_string()), None, None, None);
+                    }
                 }
-            }
-        }
-    }
-    for container in pod_spec.containers.iter_mut() {
-        match enforce_container_security_policies(container, &validation_request) {
-            Ok(mutate_request) => mutated = mutated || mutate_request,
-            Err(error) => {
-                return kubewarden::reject_request(Some(error.to_string()), None, None, None)
-            }
-        }
-    }
 
-    match enforce_supplemental_groups(pod_spec.security_context.clone(), &validation_request) {
-        Ok(mutate_security_context) => {
-            if mutate_security_context.is_some() {
-                mutated = true;
-                pod_spec.security_context = mutate_security_context;
-            }
-        }
-        Err(msg) => {
-            return kubewarden::reject_request(Some(msg.to_string()), None, None, None);
+                if mutated {
+                    let mut pod: apicore::Pod =
```

Comment:
this mutation will fail if the resource is not a Pod (e.g. Deployment). Should we just mutate the object if it is a Pod?

Reply:
Fixed.